packages/backend/src/session/service.ts

```diff
@@ -0,0 +1,78 @@
+import { BaseService } from '../shared/baseService'
+import { uuid } from '../connector/core'
+import IORedis from 'ioredis'
+import { Session } from './util'
+
+export interface SessionService {
+  create(): Promise<Session>
+  revoke(session: SessionOptions): void
+  refresh(session: SessionOptions): Promise<Session | undefined>
+  get(session: SessionOptions): Promise<Session | undefined>
+}
+
+interface ServiceDependencies extends BaseService {
+  redis: IORedis.Redis
+}
+
+type SessionOptions = {
+  key: string
+}
+
+export async function createSessionService({
+  logger,
+  redis
+}: ServiceDependencies): Promise<SessionService> {
+  const log = logger.child({
+    service: 'SessionService'
+  })
+  const deps: ServiceDependencies = {
+    logger: log,
+    redis
+  }
+  return {
+    create: () => createSession(deps),
+    revoke: (options: SessionOptions) => revokeSession(deps, options),
+    refresh: (options: SessionOptions) => refreshSession(deps, options),
+    get: (options: SessionOptions) => getSession(deps, options)
+  }
+}
+
+async function createSession(deps: ServiceDependencies): Promise<Session> {
+  const key = uuid()
+  await deps.redis.set(key, 0, 'EX', 30 * 60)
+  const sessionExpiry = await deps.redis.pttl(key)
+  return {
+    key,
+    expiresAt: new Date(Date.now() + sessionExpiry)
+  }
+}
+
+async function revokeSession(
+  deps: ServiceDependencies,
+  { key }: SessionOptions
+) {
+  deps.redis.del(key)
+}
+
+async function refreshSession(
+  deps: ServiceDependencies,
+  { key }: SessionOptions
+): Promise<Session | undefined> {
+  await deps.redis.expire(key, 30 * 60)
```

Comment:
This could check the response to know if the session doesn't exist.
Also what if here and `createSession` calculated `expiresAt` instead of calling `pttl`?
(It looks like we could use [`pexpiretime`](https://redis.io/commands/expiretime) when redis 7 is out.)

Reply:
Using `expireAt` is probably cleaner and makes testing easier, too. Thanks!

Reply:
Or I just store the unix timestamp as value in the meantime.

Reply:
Yeah, maybe `create` and `refresh` should just calculate the timestamp and pass it to:
```
deps.redis.set(key, sessionExpiry, 'PXEX', sessionExpiry)
```

and `get` can do  
```
deps.redis.get(key)
```

Then if/when there's redis 7, we can go back to an empty value.
`refresh` can call:
```
deps.redis.pexpire(key, sessionExpiry)
```
and `get` can use:
```
deps.redis.pexpiretime(key)